### PR TITLE
Fix #356: Add further support for router config (tls termination and - Incremental to #375

### LINKED ISSUE #356: Configuring router in openshift-maven-plugin
### Description
Hi

trying to configure end-to-end encryption for my deployment via the plugin. According to https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html#route-types I need to configure the router for tls with either rencrypt or passthrough, like the below example:
```
spec:
  port:
    targetPort: https
  tls:
    termination: reencrypt
    insecureEdgeTerminationPolicy: Redirect
```
I can add via a yaml fragment. However, it would be nice to not having to write any yaml at all :)

version: openshift-maven-plugin:1.0.0-rc-1

Any thoughts on this? It seems like AutoTLS is a bit moot, if the route cannot be secured? Or is there another use case for this?

### PR DESCRIPTION
**Pull request by manusa**
## Description

Incremental PR for #375 (@sejacobsen) (Closes #375 + #356 )

Added annotations so Router can be configured w/ TLS termination from the pom. Updated docs accordingly.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] ~~I tested my code in Kubernetes~~
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc [hunk 1]
```diff
@@ -226,8 +226,54 @@ If you do not want to generate a Route descriptor, you can set the `jkube.opensh
...
+
+| *termination*
+a| termination indicates termination type. The following values are supported:
+
+* edge (default)
+* passthrough
+* reencrypt
+
+See https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html#secured-routes or https://docs.openshift.com/container-platform/latest/networking/routes/secured-routes.html
+| `jkube.enricher.jkube-openshift-route.termination`
```
Comment (sejacobsen) [inline kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc:238]: should be 'tlsTermination', no?

File: kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc [hunk 2]
```diff
@@ -226,8 +226,54 @@ If you do not want to generate a Route descriptor, you can set the `jkube.opensh
...
+  <groupId>org.eclipse.jkube</groupId>
+  <artifactId>{plugin}</artifactId>
+  <version>{version}</version>
+  <configuration>
+    <enricher>
+      <config>
+        <jkube-openshift-route>
+          <generateRoute>true</generateRoute>
+          <insecureEdgeTerminationPolicy>Allow</insecureEdgeTerminationPolicy>
+          <termination>edge</termination>
```
Comment (sejacobsen) [inline kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc:267]: <tlsInsecureEdgeTerminationPolicy>Allow</tlsInsecureEdgeTerminationPolicy>
          <tlsTermination>edge</tlsTermination>

File: kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc (hunk 1 above)
Comment (manusa) [reply kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc:238]: :+1:  :facepalm:

**Review by sejacobsen — CHANGES_REQUESTED** [review]
Hi, please see my comments for suggested changes

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**